aggregator/src/state_transition.rs

```diff
@@ -155,200 +188,70 @@ pub fn state_transition_circuit<F: BigPrimeField>(
 
-    let old_root = ctx.load_witness(input.nullifier_tree.old_root);
+    let old_root = ctx.load_witness(nullifier_tree.old_root);
     let low_leaf = IndexedMerkleTreeLeaf::new(val, next_val, next_idx);
 
-    let new_root = ctx.load_witness(input.nullifier_tree.new_root);
+    let new_root = ctx.load_witness(nullifier_tree.new_root);
 
-    let val = ctx.load_witness(input.nullifier_tree.new_leaf.val);
-    assert_eq!(val.value(), nullifier_hash.value());
+    let val = ctx.load_witness(nullifier_tree.new_leaf.val);
+    //assert_eq!(val.value(), nullifier_hash.value());
     ctx.constrain_equal(&val, &nullifier_hash);
-    let next_val = ctx.load_witness(input.nullifier_tree.new_leaf.next_val);
-    let next_idx = ctx.load_witness(input.nullifier_tree.new_leaf.next_idx);
+    let next_val = ctx.load_witness(nullifier_tree.new_leaf.next_val);
+    let next_idx = ctx.load_witness(nullifier_tree.new_leaf.next_idx);
 
     let new_leaf = IndexedMerkleTreeLeaf::new(val, next_val, next_idx);
 
-    let new_leaf_index = ctx.load_witness(input.nullifier_tree.new_leaf_index);
-    let is_new_leaf_largest = ctx.load_witness(input.nullifier_tree.is_new_leaf_largest);
+    let new_leaf_index = ctx.load_witness(nullifier_tree.new_leaf_index);
+    let is_new_leaf_largest = ctx.load_witness(nullifier_tree.is_new_leaf_largest);
 
-    let low_leaf_proof = input
-        .nullifier_tree
+    let low_leaf_proof = nullifier_tree
         .low_leaf_proof
         .iter()
         .map(|x| ctx.load_witness(*x))
         .collect::<Vec<_>>();
-    let low_leaf_proof_helper = input
-        .nullifier_tree
+    let low_leaf_proof_helper = nullifier_tree
         .low_leaf_proof_helper
         .iter()
         .map(|x| ctx.load_witness(*x))
         .collect::<Vec<_>>();
-    let new_leaf_proof = input
-        .nullifier_tree
+    let new_leaf_proof = nullifier_tree
         .new_leaf_proof
         .iter()
         .map(|x| ctx.load_witness(*x))
         .collect::<Vec<_>>();
-    let new_leaf_proof_helper = input
-        .nullifier_tree
+    let new_leaf_proof_helper = nullifier_tree
         .new_leaf_proof_helper
         .iter()
         .map(|x| ctx.load_witness(*x))
         .collect::<Vec<_>>();
 
-    insert_leaf::<F, 3, 2>(
-        ctx,
-        range,
-        &hasher,
-        &old_root,
-        &low_leaf,
-        &low_leaf_proof,
-        &low_leaf_proof_helper,
-        &new_root,
-        &new_leaf,
-        &new_leaf_index,
-        &new_leaf_proof,
-        &new_leaf_proof_helper,
-        &is_new_leaf_largest,
-    );
-
-    // PK_ENC N
-    public_inputs.extend(pk_enc.n.limbs());
-
-    // PK_ENC G
-    public_inputs.extend(pk_enc.g.limbs());
-
-    // PREV_VOTE
-    for enc_vote in prev_vote {
-        public_inputs.extend(enc_vote.limbs());
-    }
-
-    // INCOMING_VOTE
-    for enc_vote in incoming_vote {
-        public_inputs.extend(enc_vote.limbs());
-    }
+    // insert_leaf::<F, 3, 2>(
```

Comment:
@Vishalkulkarni45. it should be un-commented out right?

Reply:
yes